pkg/config/config.go

```diff
@@ -430,12 +419,19 @@ func load(fs *flag.FlagSet, args []string, loader loaderFunc) (*Config, error) {
 
 // CheckSecret is a helper function to ensure the original value is overwritten with <secret>
 func CheckSecret(t *testing.T, rawCfg string, originalValue string) {
-	var cfg = &Config{}
-	err := LoadBytes([]byte(rawCfg), false, cfg)
+	var cfg Config
+	err := LoadBytes([]byte(rawCfg), false, &cfg)
 	require.NoError(t, err)
-	bb, err := yaml.Marshal(cfg)
+
+	// Set integrations version to make sure our marshal function goes through
+	// the custom marshaling code.
+	err = cfg.Integrations.setVersion(integrationsVersion1)
 	require.NoError(t, err)
```

Comment:
I'm genuinely not sure why this worked before. The version needs to be set for remarshaling to work properly.